File: src/org/elixir_lang/ElixirFileType.java
package org.elixir_lang;

import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

/**
 * Created by luke.imhoff on 7/27/14.
 */
public class ElixirFileType extends LanguageFileType {
    public static final ElixirFileType INSTANCE = new ElixirFileType();

    protected ElixirFileType() {
        super(ElixirLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "Elixir";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "Elixir language file";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "ex";
    }

    @NotNull
    @Override
    public Icon getIcon() {
        return Icons.FILE;
    }
}

File: src/org/elixir_lang/ElixirFileType.kt
package org.elixir_lang

import com.intellij.openapi.fileTypes.LanguageFileType
import javax.swing.Icon

/**
 * Created by luke.imhoff on 7/27/14.
 */
open class ElixirFileType protected constructor() : LanguageFileType(ElixirLanguage) {
    override fun getName(): String = "Elixir"
    override fun getDescription(): String = "Elixir language file"
    override fun getDefaultExtension(): String = "ex"
    override fun getIcon(): Icon = Icons.FILE

    companion object {
        @JvmField
        val INSTANCE = ElixirFileType()
    }
}
